src/core/drive/page_renderer.ts

```diff
@@ -1,19 +1,26 @@
 import { Renderer } from "../renderer"
 import { PageSnapshot } from "./page_snapshot"
 import { ReloadReason } from "../native/browser_adapter"
-import { activateScriptElement, waitForLoad } from "../../util"
+import { activateScriptElement, waitForLoad, nextEventLoopTick, getBodyElementId } from "../../util"
 
 export class PageRenderer extends Renderer<HTMLBodyElement, PageSnapshot> {
-  static renderElement(currentElement: HTMLBodyElement, newElement: HTMLBodyElement) {
+  static async renderElement(currentElement: HTMLBodyElement, newElement: HTMLBodyElement) {
+    await nextEventLoopTick()
+
     if (document.body && newElement instanceof HTMLBodyElement) {
-      document.body.replaceWith(newElement)
+      const bodyElementId = getBodyElementId()
+
```

Comment:
Thanks for the response @agrobbin! I think we can even get rid of this line here
```suggestion

```